extensions/control-plane/transfer/transfer-data-plane-signaling/src/main/java/org/eclipse/edc/connector/controlplane/transfer/dataplane/flow/DataPlaneSignalingFlowController.java

```diff
@@ -97,46 +99,66 @@ public boolean canHandle(TransferProcess transferProcess) {
                 .properties(propertiesResult.getContent())
                 .build();
 
-        var dataPlaneInstanceId = dataPlaneInstance != null ? dataPlaneInstance.getId() : null;
-
-        return clientFactory.createClient(dataPlaneInstance)
-                .start(dataFlowRequest)
-                .map(it -> DataFlowResponse.Builder.newInstance()
-                        .dataAddress(it.getDataAddress())
-                        .dataPlaneId(dataPlaneInstanceId)
-                        .build()
-                );
+        var selection = selectorClient.select(transferProcess.getContentDataAddress(), transferProcess.getTransferType(), selectionStrategy);
+        if (selection.succeeded()) {
+            var dataPlaneInstance = selection.getContent();
+            return clientFactory.createClient(dataPlaneInstance)
+                    .start(dataFlowRequest)
+                    .map(it -> DataFlowResponse.Builder.newInstance()
+                            .dataAddress(it.getDataAddress())
+                            .dataPlaneId(dataPlaneInstance.getId())
+                            .build()
+                    );
+        } else {
+            // TODO: this branch works for embedded data plane but it is a potential false positive when the dataplane is not found, needs to be refactored
+            return clientFactory.createClient(null)
```

Comment:
shouldn't this just return a `failure`? It doesn't really matter whether no data plane exists to serve the request, or no data plane is _reachable_ that could serve the request.

Reply:
needs to work when the data plane is embedded, that's why I left a comment, it is covered by e2e tests, when the dataplane is embedded the selector returns null (so not-found), already on my todo list.

Reply:
ok good then.